### ISSUE TITLE
Fuse doesn't use linux/mac user to read/write

### ISSUE THREAD

**Issue by LuQQiu** [Alluxio/alluxio#14838]
**Alluxio Version:**
What version of Alluxio are you using?

**Describe the bug**
Fuse doesn't use linux/mac user to read/write
To be tested

**To Reproduce**
Steps to reproduce the behavior (as minimally and precisely as possible)

**Expected behavior**

**Urgency**
Describe the impact and urgency of the bug.

**Are you planning to fix it**
Please indicate if you are already working on a PR. 

**Additional context**

**Comment by LuQQiu** [comment]
From community user:
访问fuse路径下面的东西，都会使用你启动fuse的用户身份，而不是你的linux用户，如果在本地访问fuse的用户都是fuse的启动用户还好，但是我本地是允许所有用户访问的，这样会造成歧义，比如一个非启动fuse进程的用户他其实没有权限访问fuse下面的某个文件，但是启动fuse的用户有权限，那么他是可以访问到这个文件的

alluxio.fuse.user.group.translation.enabled 没用的，设置这个只是文件显示能看到所属不同的用户，不设这个看到所有文件的用户都是启动用户，但是真正判断你是否可以读的，还是用fuse的启动用户，无论你在本地用哪个用户访问

**Comment by LuQQiu** [comment]
Alluxio uses https://github.com/Alluxio/alluxio/blob/master/core/common/src/main/java/alluxio/security/authentication/AuthenticatedClientUser.java
to get the gRPC user group (who issues the gRPC call).
This is correct when client connects to master using gRPC but it's not correct in Fuse case.
In Fuse case, client will always be the client who launches Fuse application instead of the linux user who performs the open(), create() calls.
Need a comprehensive fuse permission story to deal with those clients.

**Comment by LuQQiu** [comment]
@yyongycy FYI, this will need a more thorough discussion

**Comment by yyongycy** [comment]
That is right. Is there any specific requirement? And what is the expectation? 
In general, the posix is using the user of linux/mac users, that is not aligned with any network user.

**Comment by LuQQiu** [comment]
The expectation is that Fuse performs similar to Alluxio client.
If user A can read/write to Alluxio through Alluxio shell, then it can read/write to Alluxio through Fuse. If A cannot read/write to Alluxio through Alluxio shell, then it cannot read/write to Alluxio through Fuse.

**Comment by LuQQiu** [comment]
Alluxio client-side impersonation may help solve the issue
https://www.alluxio.io/blog/alluxio-developer-tip-why-am-i-seeing-the-error-user-yarn-is-not-configured-for-any-impersonation-impersonationuser-foo/

**Comment by LuQQiu** [comment]
Hadoop impersonation may not fit Fuse case.
Hadoop impersonation is 
one Haodoop client -> one Hadoop filesystem -> one Alluxio client -> one Alluxio filesystem
while in fuse
we use one Alluxio filesystem but with unknown amount of Fuse users.
It's not easy to use the existing hadoop impersonation.

**Comment by LuQQiu** [comment]
Feedback from community user
```
$ integration/fuse/bin/alluxio-fuse mount -o allow_other -o default_permissions /mnt/alluxio/ /
我先用qw用户用这个命令启动fuse之后，再用hdfs账户根本进不去/mnt/alluxio目录了，现在hdfs用户看到的目录权限变成了
hdfs@node08:/gxr/wangqian/alluxio-2.7.1$ ll /mnt/
ls: cannot access '/mnt/alluxio': Permission denied
total 16
drwxr-xr-x 5 root root 4096 12月 28 14:35 ./
drwxr-xr-x 22 root root 4096 12月 10 15:50 ../
d????????? ? ?   ?      ?            ? alluxio/
这个目录在hdfs挂载的权限是hdfs:hdfs，用qw账户看是正常的
$ ll /mnt/alluxio
ls: cannot access '/mnt/alluxio': Permission denied

如果不带上default_permissions，用hdfs账户可以进入/mnt/alluxio，只不过一切操作其实是用qw验证的（因为是qw启动的fuse），如果qw没有权限的文件，hdfs照样看不了，即使这个文件在hdfs本身就是hdfs:hdfs的权限
```
